### DISCUSSION TITLE
Supabase auth.api.createUser not sending verification email.

### DISCUSSION THREAD

**Discussion by jishnair** [supabase/supabase#5260] [Questions]
I have tried to create a user from the server using the  **auth.api.createUser** API  and doesn't seems to send the confirmation email. With the **supabase.auth.signUp**  the email confirmations are working. Can some one confirm if it's a bug or am doing something wrong ?

**Comment by GaryAustin1** [comment]
Not clear if Supabase supports this gotrue option or not for createUser() ... { email:email, password:password,  confirm: true}  
GoTrue requires the confirm option set to send the email confirmation.

**Reply by lockinmarv** [reply]
That's actually not true. It has a `confirm` option but it's what `email_confirm` is in Supabase. This option just skips the verification and sets the user as verified in the database. It's not for sending the email.

**Comment by cbertram** [comment]
I am experiencing this problem too. Did you find a solution, @jishnair?

**Comment by J0** [comment]
Hey all,

@m4rvr  is right, the intended behavior is that `createUser` with the confirmation option set does not send an email verification -- it only sets the user's verified status to true.

**Reply by hassannteifeh** [reply]
Seems like a broken flow to me. If `email_confirm` is set to `false`(which is the default value), then the dashboard will show status "waiting for verification", but how exactly is verification supposed to happen without supabase sending the confirmation email?

**Reply by hassannteifeh** [reply]
Moreover, if in both cases (false and true), no action is taken, then maybe it's better to use something like `user_is_confirmed` instead. `email_confirm` sounds to me like something that would trigger an action.

**Comment by cbertram** [comment]
For me the solution was simply to use `supabase.auth.signUp` from the server, if I want the confirmation email to be sent.

It seems confusing to me, that `auth.api.createUser` doesn't send a confirmation email, if `email_confirm` is false, though. There doesn't seem to be any way to generate a confirmation link manually with `auth.api.generateLink`. (Maybe there is.) If this is how things are intended to work, maybe a sentence should be added to the documentation of `createUser`?

**Reply by hassannteifeh** [reply]
I usually don't use a function or a method on the backend that is intended to be used on the client side, unless I know how that function exactly works. 

To begin with it makes you wonder if the method/function is internally stateful, which could share state across requests or use caching/memoization.

In addition to that, not being able to differentiate between requests coming from the backend server and those from the frontend makes it much harder for the system to accurately evaluate spam/abuse of the system (e.g 50 signup requests coming from the server vs 50 signup requests coming from a single website/app).
